src/lava/proc/monitor/models.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+from lava.magma.core.model.py.ports import PyRefPort, PyVarPort, PyInPort
+from lava.proc.monitor.process import Monitor
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.resources import CPU
+
+
+# A minimal PyProcModel implementing P1
+@implements(proc=Monitor, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('floating_pt')
+class PyMonitorModel(PyLoihiProcessModel):
+    """
+    This process model contains prototypical Ports and Vars to have
+    one-to-one correspondes with Monitor process.
+    """
+    var_read_0: np.ndarray = LavaPyType(np.ndarray,
+                                        np.float,
+                                        precision=24)
+    out_read_0: np.ndarray = LavaPyType(np.ndarray,
+                                        np.float,
+                                        precision=24)
+    ref_port_0: PyRefPort = LavaPyType(PyRefPort.VEC_DENSE, int)
+    in_port_0: PyInPort = LavaPyType(PyInPort.VEC_DENSE, int)
+
+
+    def pre_guard(self):
+        return True
+
+    def run_pre_mgmt(self):
+        """
+        During this phase, RefPorts of Monitor process collects data from
+        monitored Vars
+        """
+        # check if this Monitor Process instance has been assigned to monitor
+        # any Var by checking proc_params["n_ref_ports"]
+        if self.current_ts > 0 and self.proc_params["n_ref_ports"] > 0:
```

Comment:
This if seems unnecessary as self.current_ts should be always > 0 and you check in the for loop if there are any ref ports.

Reply:
Sounds good